lib/remote-connector.js

```diff
@@ -70,16 +95,29 @@ RemoteConnector.prototype.define = function(definition) {
 };
 
 RemoteConnector.prototype.resolve = function(Model) {
-  var remotes = this.remotes;
+  var auth = this.auth,
+      remotes = this.remotes,
+      headers = this.headers,
+      headerArgs = this.headerArgs;
 
   Model.sharedClass.methods().forEach(function(remoteMethod) {
     if (remoteMethod.name !== 'Change' && remoteMethod.name !== 'Checkpoint') {
-      createProxyMethod(Model, remotes, remoteMethod);
+      headers.forEach(function(header) {
+        var already = remoteMethod.accepts.filter(function(accept) {
+          return accept.http &&
+              accept.http.source === 'header' &&
+              accept.name === header.name;
+        });
+        if (!already || already.length === 0) {
+          remoteMethod.accepts.push(header);
```

Comment:
> Is the request object not guaranteed to be there?

Actually I was mistaken. I think you added this:

https://github.com/strongloop/strong-remoting/blob/master/lib/rest-adapter.js#L117

So it should be available!